collector/lib/CollectorConfig.cpp

```diff
@@ -400,6 +403,56 @@ void CollectorConfig::HandleSinspEnvVars() {
   }
 }
 
+bool CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
+  if (yamlConfig.IsNull() || !yamlConfig.IsDefined()) {
+    CLOG(FATAL) << "Unable to read config from config file";
+    return false;
+  }
+  YAML::Node networkConnectionConfig = yamlConfig["networkConnectionConfig"];
+  if (!networkConnectionConfig) {
+    CLOG(WARNING) << "No networkConnectionConfig in config file";
+    return false;
+  }
+
+  bool enableExternalIps = false;
+  if (networkConnectionConfig["enableExternalIps"]) {
+    enableExternalIps = networkConnectionConfig["enableExternalIps"].as<bool>(false);
+  }
+
+  sensor::CollectorConfig runtime_config;
+  auto* networkConfig = runtime_config.mutable_network_connection_config();
+  networkConfig->set_enable_external_ips(enableExternalIps);
+
+  SetRuntimeConfig(runtime_config);
+  CLOG(INFO) << "Runtime configuration:";
+  CLOG(INFO) << GetRuntimeConfigStr();
+
+  return true;
+}
+
+void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
+  if (!std::filesystem::exists(filePath)) {
+    CLOG(DEBUG) << "No configuration file found. " << filePath;
+    return;
+  }
+
+  try {
+    YAML::Node yamlConfig = YAML::LoadFile(filePath);
+    YamlConfigToConfig(yamlConfig);
+  } catch (const YAML::BadFile& e) {
```

Comment:
As I understand it, at one point, we will call this several times after startup to re-read the configuration.
I just want to underline that in this case, re-reading and consuming the config will likely happen in different threads and we should guard against race condition.

It is probably fine for now if we are calling `HandleConfig()` only at startup. Do we want to implement locking now or commit to do it in a subsequent PR ?

Reply:
I've actually discussed this with @Stringy when he was some of these things, we need to start reading configuration from a centralized static `CollectorConfig` and a `std::shared_mutex` will need to be put in place so we can lock it exclusively to update configuration and shared for only reading it. It'd be nice if the mechanism is broad enough to lock the entire configuration, that way we can start moving everything to the configuration file.

For now, I would go with the existing approach, since there is no actual updating going on while running and change it in a follow up, as long as we are all aware this is a thing we need to do we should be fine. 

Good call out though!

Reply:
Let's implement locking when we actually need it.